financialaid/serializers.py

```diff
@@ -110,53 +113,70 @@ def save(self):
 class FinancialAidActionSerializer(serializers.Serializer):
     """
     Serializer for financial aid actions
+
+    Note: FinancialAidStatus.REJECTED has been temporarily disabled as a valid status for FinancialAid objects. While
+    the functions/code that handle this case have been left intact, this option has been disabled by removing the
+    status from the choices for "action" on this serializer as well as from the valid choices on the database model.
     """
     action = ChoiceField(
         choices=[
-            FinancialAidStatus.REJECTED,
+            # FinancialAidStatus.REJECTED,
             FinancialAidStatus.APPROVED,
             FinancialAidStatus.PENDING_MANUAL_APPROVAL
         ],
         write_only=True
     )
     tier_program_id = IntegerField(write_only=True)
+    justification = ChoiceField(
+        choices=FinancialAidJustification.ALL_JUSTIFICATIONS,
+        default=None,
+        write_only=True
+    )
 
     def validate(self, data):
         """
         Validators for this serializer
         """
-        # Check that the previous financial aid status allows for the new status
-        if (data['action'] == FinancialAidStatus.REJECTED and
-                self.instance.status != FinancialAidStatus.PENDING_MANUAL_APPROVAL):
-            raise ValidationError("Cannot reject application that is not pending manual approval.")
-        if (data['action'] == FinancialAidStatus.APPROVED and
-                self.instance.status != FinancialAidStatus.PENDING_MANUAL_APPROVAL):
-            raise ValidationError("Cannot approve application that is not pending manual approval.")
-        if (data['action'] == FinancialAidStatus.PENDING_MANUAL_APPROVAL and
-                self.instance.status not in [FinancialAidStatus.PENDING_DOCS, FinancialAidStatus.DOCS_SENT]):
-            raise ValidationError("Cannot mark documents as received for an application awaiting docs.")
-        # Check tier program exists
-        try:
-            data["tier_program"] = TierProgram.objects.get(
-                id=data["tier_program_id"],
-                program_id=self.instance.tier_program.program_id,
-                current=True
-            )
-        except TierProgram.DoesNotExist:
-            raise ValidationError("Financial Aid Tier does not exist for this program.")
+        # Required field
+        if data.get("action", None) is None:
+            raise ValidationError({"action": "This field is required."})
+        # For approving and rejecting
+        if data["action"] in [FinancialAidStatus.APPROVED, FinancialAidStatus.REJECTED]:
+            # Required fields
+            if data.get("tier_program_id", None) is None:
+                raise ValidationError({"tier_program_id": "This field is required."})
+            if data.get("justification", None) is None:
+                raise ValidationError({"justification": "This field is required."})
```

Comment:
These are used on the frontend.

The dicts vs strings is how DRF differentiates between field errors and non-field errors.